src/core/egg-herbie.rkt

```diff
@@ -1245,6 +1247,26 @@
               itypes))]
     [(list _ ...) +inf.0])) ; specs
 
+;; Old cost model version
+(define (default-egg-cost-proc regraph cache node type rec)
+  (match node
+    [(? number?) 1]
+    [(? symbol?) 1]
+    [(list 'if cond ift iff)
+     (+ 1 (rec cond (get-representation 'bool) 1)
+          (rec ift type 1)
+          (rec iff type 1))]
```

Comment:
Failure values should be `+inf.0` to prevent extraction

Reply:
Fixing